src/interface/sparse_pack.hpp

```diff
@@ -183,73 +183,133 @@ class SparsePack : public SparsePackBase {
     Get(pmd, vars, flags, fluxes, coarse);
   }
 
+  template <class T>
+  static SparsePack GetFlat(T *pmd, const std::vector<MetadataFlag> &flags = {}) {
+    const bool coarse = false;
+    const bool fluxes = false;
+    const bool flatten = true;
+    return Get(pmd, flags, fluxes, coarse, flatten);
+  }
+
+  template <class T, class VAR_VEC>
+  static std::tuple<SparsePack, SparsePackIdxMap>
+  GetFlat(T *pmd, const VAR_VEC &vars, const std::vector<MetadataFlag> &flags = {}) {
+    const bool coarse = false;
+    const bool fluxes = false;
+    const bool flatten = true;
+    return Get(pmd, vars, flags, fluxes, coarse, flatten);
+  }
+
+  template <class T>
+  static SparsePack GetFlatWithFluxes(T *pmd,
+                                      const std::vector<MetadataFlag> &flags = {}) {
+    const bool coarse = false;
+    const bool fluxes = true;
+    const bool flatten = true;
+    return Get(pmd, AddFlag_(flags), fluxes, coarse, flatten);
+  }
+
+  template <class T, class VAR_VEC>
+  static std::tuple<SparsePack, SparsePackIdxMap>
+  GetFlatWithFluxes(T *pmd, const VAR_VEC &vars,
+                    const std::vector<MetadataFlag> &flags = {}) {
+    const bool coarse = false;
+    const bool fluxes = true;
+    const bool flatten = true;
+    return Get(pmd, vars, AddFlag_(flags), fluxes, coarse, flatten);
+  }
+
+  template <class T>
+  static SparsePack GetFlatWithCoarse(T *pmd,
+                                      const std::vector<MetadataFlag> &flags = {}) {
+    const bool coarse = true;
+    const bool fluxes = false;
+    const bool flatten = true;
+    return Get(pmd, flags, fluxes, coarse, flatten);
+  }
+
+  template <class T, class VAR_VEC>
+  static std::tuple<SparsePack, SparsePackIdxMap>
+  GetFlatWithCoarse(T *pmd, const VAR_VEC &vars,
+                    const std::vector<MetadataFlag> &flags = {}) {
+    const bool coarse = true;
+    const bool fluxes = false;
+    const bool flatten = true;
+    return Get(pmd, vars, flags, fluxes, coarse, flatten);
+  }
+
   // Methods for getting parts of the shape of the pack
   KOKKOS_FORCEINLINE_FUNCTION
   int GetNBlocks() const { return nblocks_; }
 
-  KOKKOS_FORCEINLINE_FUNCTION
-  int GetNDim() const { return ndim_; }
-
   KOKKOS_FORCEINLINE_FUNCTION
   int GetMaxNumberOfVars() const { return pack_.extent_int(2); }
 
   KOKKOS_INLINE_FUNCTION
-  const Coordinates_t &GetCoordinates(const int b) const { return coords_(b)(); }
+  const Coordinates_t &GetCoordinates(const int b = 0) const { return coords_(b)(); }
 
   // Bound overloads
-  KOKKOS_INLINE_FUNCTION int GetLowerBound(const int b) const { return 0; }
+  KOKKOS_INLINE_FUNCTION int GetLowerBound(const int b = 0) const { return 0; }
```

Comment:
I think this should change for flat packs. `GetLowerBound(b)` with `b > 0` should return `bounds_(0, b, 0)` and `GetUpperBound(b)` should return `bounds_(1, b, nvar_)`.